tools/pythonpkg/duckdb-stubs/__init__.pyi

```diff
@@ -161,13 +161,13 @@ class DuckDBPyResult:
     def arrow(self, chunk_size: int = ...) -> pyarrow.lib.Table: ...
     def close(self) -> None: ...
     def description(self) -> list: ...
-    def df(self) -> pandas.DataFrame: ...
+    def df(self, date_as_datetime: bool) -> pandas.DataFrame: ...
     def fetch_arrow_reader(self, approx_batch_size: int) -> pyarrow.lib.RecordBatchReader: ...
     def fetch_arrow_table(self, chunk_size: int = ...) -> pyarrow.lib.Table: ...
-    def fetch_df(self) -> pandas.DataFrame: ...
-    def fetch_df_chunk(self, num_of_vectors: int = ...) -> pandas.DataFrame: ...
+    def fetch_df(self, date_as_datetime: bool) -> pandas.DataFrame: ...
```

Comment:
Boolean parameters are in general a bit dangerous - how do other systems (e.g. arrow) do this? Perhaps having a global setting is more readable/clear?

Reply:
Arrow has a boolean as well (https://arrow.apache.org/docs/python/pandas.html#date-types)

I also think global is cleaner, but on the other hand, is good to follow stds. If you strongly prefer global I'm fine going with it.

Reply:
Can we somehow enforce this to be a named parameter, so you would e.g. have to type `fetch_df(date_as_datetime=True)`?

Should we also use the same naming (i.e. `date_as_object`)?

Reply:
You can yeah, like this:

```py
 def fetch_df(self, *, date_as_datetime: bool)
```
Not sure if pybind11 supports that tho

Reply:
I think you had a good point before as well, it seems to me a user would either emit all dfs one way or the other, so maybe a global config is cleaner than adding this type-specific option to ```df()```.
If no one is strongly opposed to that, I think I'll go that way.